File: 100-199/172-factorial-trailing-zeroes.py
# coding: utf-8

class Solution:
    # @return an integer
    def trailingZeroes(self, n):
    	if n/5:
    		return n/5 + self.trailingZeroes(n/5)
    	else:
    		return 0

a = Solution()
print a.trailingZeroes(100)

File: special-1-to-200/172-factorial-trailing-zeroes.go
package main

func trailingZeroes(n int) int {
	// 只需统计 [1, n] 中，5 的个数
	// 想象一个金字塔
	// 第一层中，n 的个数是 n / 5
	// 第二层中，是 n / 25
	// 然后累加起来
	zeroCount := 0
	for n > 0 {
		n /= 5
		zeroCount += n
	}
	return zeroCount
}
